docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd

```diff
@@ -0,0 +1,180 @@
+
+```{python}
+from pathlib import Path
+import tarfile
+from urllib.request import urlretrieve
+
+#Download data
+url = "https://github.com/iamgeoknight/common_datasets/raw/main/parquet/202003-citibike-tripdata.tar.xz"
+tar_path = Path("202003-citibike-tripdata.tar.xz")
+parquet_path = Path("202003-citibike-tripdata.parquet")
+
+if not tar_path.exists():
+    urlretrieve(url, tar_path)
+
+if not parquet_path.exists():
+    with tarfile.open(tar_path, 'r:xz') as t:
+        t.extract("202003-citibike-tripdata.parquet")
+```
+
+Now that we have the data, we import Ibis and turn on the interactive mode, to
+easily explore the output of our queries.
+
+```{python}
+import ibis
+from ibis import _
+from ibis.interactive import *
+```
+
+## Let's get started
+
+Because this dataset does not contain any geometries, we have to load the spatial extension. If the dataset included any geometry columns, Ibis is smart enough to load the extension for us upon reading the data.
+
+```{python}
+con = ibis.duckdb.connect("biketrip.ddb")
+con.load_extension("spatial")
+
+#read data and rename columns to use snake case
+biketrip = con.read_parquet("202003-citibike-tripdata.parquet").rename("snake_case")
+biketrip
+```
+
+
+We have the information about the longitude and latitude for start and end stations,
+to create geometry points and put the spatial features to use.
+
+## Create bike trip table
+
+In the original tutorial, Spatial Dev Guru creates a table with transformed
+"Pickup" and "Dropoff" points. In DuckDB the `st_transform` function takes by default
+points as `YX` (latitude/longitude) while in Ibis,  we assume data in the form
+`XY` (longitude/latitude) to be consistent with PostGIS and Geopandas.
+
+In Ibis the function to tr
+```{python}
+# Notice longitude/latitude order
+pickup = biketrip.start_station_longitude.point(biketrip.start_station_latitude).name("pickup")
+dropoff = biketrip.end_station_longitude.point(biketrip.end_station_latitude).name("dropoff")
+
+#convert is the equivalent of `st_transform`
+biketrip = biketrip.mutate(pickup_point=pickup.convert('EPSG:4326', 'EPSG:3857'),
+                           dropoff_point=dropoff.convert('EPSG:4326', 'EPSG:3857')
+                        )
+
+biketrip[["pickup_point", "dropoff_point"]]
+```
+
+Using `mutate` we add two new columns to our `biketrip` table with transformed
+spatial points for pickup and dropoff locations, that are in the Web Mercator projection ([EPSG:3857](https://epsg.io/3857)).
+
+## Identify popular starts and end stations
+
+The following queries retrieve a list of bike start and end stations with their respective trip count in descending order.
+
+**Top 10 start stations by trip count**
+
+```{python}
+biketrip.group_by(biketrip.start_station_name).agg(trips = ibis._.count()).order_by(ibis.desc("trips"))
+```
+
+**Top 10 end stations by trip count**
+
+```{python}
+biketrip.group_by(biketrip.end_station_name).agg(trips = ibis._.count()).order_by(ibis.desc("trips"))
```

Comment:
```suggestion
biketrip.end_station_name.topk(10, by=_.trips.count())
```

Might work, similar to the `start_station_name` variant.

Reply:
nice, I wasn't aware of the `topk()` function, the equivalent query would be
`biketrip.end_station_name.topk(10)` In the first case I left both options to showcase the similarity.